docs/content/DimensionSpecs.md

```diff
@@ -154,3 +164,35 @@ Example for the `__time` dimension:
   "function" : "function(t) { return 'Second ' + Math.floor((t % 60000) / 1000); }"
 }
 ```
+
+### Explicit lookup extraction function
+Explicit lookups allow you to specify a set of keys and values to use when performing the extraction
+```json
+{
+  "type":"lookup",
+  "lookup":{
+    "type":"map",
+    "map":{"foo":"bar", "baz":"bat"}
+  },
+  "retainMissingValue":true,
+  "injective":true
+}
+```
+
+```json
+{
+  "type":"lookup",
+  "lookup":{
+    "type":"map",
+    "map":{"foo":"bar", "baz":"bat"}
+  },
+  "retainMissingValue":false,
+  "injective":false,
+  "replaceMissingValueWith":"MISSING"
+}
+```
+A property of `retainMissingValue` and `replaceMissingValueWith` can be specified at query time to hint how to handle missing values. Setting `replaceMissingValueWith` to `""` has the same effect of setting it to `null` or omitting the property. Setting `retainMissingValue` to true will use the dimension's original value if it is not found in the namespace. The default values are `replaceMissingValueWith = null` and `retainMissingValue = false` which causes missing values to be treated as missing.
```

Comment:
since JSON doesn't allow null keys in objects, we could use empty string to define what null values map to.